src/devices/bus/spectrum/plusd.cpp

```diff
@@ -0,0 +1,374 @@
+{
+	WD1772(config, m_fdc, 16_MHz_XTAL / 2);
+	FLOPPY_CONNECTOR(config, "fdc:0", plusd_floppies, "35dd", spectrum_plusd_device::floppy_formats).enable_sound(true);
+	FLOPPY_CONNECTOR(config, "fdc:1", plusd_floppies, "35dd", spectrum_plusd_device::floppy_formats).enable_sound(true);
+	
+	/* printer port */
+	CENTRONICS(config, m_centronics, centronics_devices, "printer");
+	m_centronics->busy_handler().set(FUNC(spectrum_plusd_device::busy_w));
+	
+	/* software list */
+	SOFTWARE_LIST(config, "flop_list").set_original("spectrum_mgt_flop");
+}
+
+
+const tiny_rom_entry *spectrum_plusd_device::device_rom_region() const
+{
+	return ROM_NAME(plusd);
+}
+
+
+//**************************************************************************
+//  LIVE DEVICE
+//**************************************************************************
+
+//-------------------------------------------------
+//  spectrum_plusd_device - constructor
+//-------------------------------------------------
+
+spectrum_plusd_device::spectrum_plusd_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
+	: device_t(mconfig, SPECTRUM_PLUSD, tag, owner, clock)
+	, device_spectrum_expansion_interface(mconfig, *this)
+	, m_rom(*this, "rom")
+	, m_fdc(*this, "fdc")
+	, m_floppy(*this, "fdc:%u", 0)
+	, m_centronics(*this, "centronics")
+{
+}
+
+//-------------------------------------------------
+//  device_start - device-specific startup
+//-------------------------------------------------
+
+void spectrum_plusd_device::device_start()
+{
+	memset(m_ram, 0, sizeof(m_ram));
+
+	save_item(NAME(m_romcs));
+	save_item(NAME(m_ram));
+	save_item(NAME(m_centronics_busy));
+}
+
+//-------------------------------------------------
+//  device_reset - device-specific reset
+//-------------------------------------------------
+
+void spectrum_plusd_device::device_reset()
+{
+	m_romcs = 0;
+	m_centronics_busy = false;
+}
+
+
+//**************************************************************************
+//  IMPLEMENTATION
+//**************************************************************************
+
+READ_LINE_MEMBER(spectrum_plusd_device::romcs)
+{
+	return m_romcs;
+}
+
+void spectrum_plusd_device::pre_opcode_fetch(offs_t offset)
+{
+	if (!machine().side_effects_disabled())
+	{
+		switch (offset)
+		{
+		case 0x0000:
+		case 0x0008:
+		case 0x003a:
+		case 0x0066:
+			m_romcs = 1;
+			break;
+		}
+	}
+}
+
+uint8_t spectrum_plusd_device::iorq_r(offs_t offset)
+{
+	uint8_t data = 0xff;
+
+	if (!machine().side_effects_disabled())
+	{
+		switch (offset & 0xff)
+		{
+		case 0xe3: // fdc status reg
+			data = m_fdc->read(0);
+			break;
+		case 0xeb: // fdc track reg
+			data = m_fdc->read(1);
+			break;
+		case 0xf3: // fdc sector reg
+			data = m_fdc->read(2);
+			break;
+		case 0xfb: // fdc data reg
+			data = m_fdc->read(3);
+			break;
+		case 0xe7: // page in
+			m_romcs = 1;
+			break;
+		case 0xf7: // bit 7: printer busy
+			data = m_centronics_busy << 7;
```

Comment:
This will cause a "using bool in int context" warning in Coverity.  Either store the actual value that will be read as a `uint8_t` in `m_centronics_busy` and use it directly, or do something  like `data = m_centronic_busy ? 0x80 : 0x00` here.